From ca65b2792d18fad6e6f16c8280e49c73a69b54d7 Mon Sep 17 00:00:00 2001
From: Kazunori Kobayashi <kkobayas@igel.co.jp>
Date: Thu, 31 May 2012 10:58:00 +0900
Subject: [PATCH 27/97] ext/dfbvideosink: set the CbCr plane address on
 semi-planar YUV buffers

is_ycbcr_planar() returns true only if a color format is planar.
It has been replaced with is_ycbcr() because the address of CbCr
plane also must be set in semi planar formats. In order to omit packed
YUV, a test to determind whether or not the format is UYVY was added.
---
 ext/directfb/dfbvideosink.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/ext/directfb/dfbvideosink.c b/ext/directfb/dfbvideosink.c
index 2dd0001..f5d07d7 100644
--- a/ext/directfb/dfbvideosink.c
+++ b/ext/directfb/dfbvideosink.c
@@ -1874,7 +1874,7 @@ gst_dfbvideosink_show_frame (GstBaseSink * bsink, GstBuffer * buf)
       goto swrender;
     }
     viosurface[0].py = GST_BUFFER_DATA (buf);
-    if (is_ycbcr_planar (viosurface[0].format))
+    if (is_ycbcr (viosurface[0].format) && viosurface[0].format != REN_UYVY)
       viosurface[0].pc = viosurface[0].py + src_pitch * src.h;
     else
       viosurface[0].pc = 0;
@@ -1899,7 +1899,7 @@ gst_dfbvideosink_show_frame (GstBaseSink * bsink, GstBuffer * buf)
       goto swrender;
     }
     viosurface[1].py = data;
-    if (is_ycbcr_planar (viosurface[1].format))
+    if (is_ycbcr (viosurface[1].format) && viosurface[0].format != REN_UYVY)
       viosurface[1].pc = viosurface[1].py + dest_pitch * result.h;
     else
       viosurface[1].pc = 0;
-- 
1.7.10.4

